apps/src/lib/levelbuilder/lesson-editor/programmingExpressionsEditorRedux.js

```diff
@@ -39,6 +57,14 @@ export default function programmingExpressions(state = [], action) {
         action.type
       );
       return action.programmingExpressions;
+    case ADD_PROGRAMMING_EXPRESSION: {
+      validateProgrammingExpression(
+        action.newProgrammingExpression,
+        action.type
+      );
+      newState = newState.concat([action.newProgrammingExpression]);
```

Comment:
I don't think we need to avoid mutating `newState`; it's a clone of the existing state object, not a reference to it

Reply:
The tests did not pass when I updated the line to just be `newState.concat([action.newProgrammingExpression]);`. Was there something else you meant?

Reply:
right, because `concat` just returns, it doesn't make any in-place changes. `newState.push(action.newProgrammingExpression)` is what you want